client/src/screens/events/Detail.js

```diff
@@ -37,9 +38,11 @@ const LONGITUDE_DELTA = LATITUDE_DELTA * ASPECT_RATIO;
 
 
 class Detail extends Component {
-  static navigationOptions = {
+  static navigationOptions = ({ navigation }) => ({
     title: 'Event Detail',
-  };
+    headerRight: <ButtonNavBar onPress={() => navigation.state.params.handleThis()} icon="mi-my-location" />,
```

Comment:
try this. if it doesn't work you need to go bind the function that is passed here
```
headerRight: <ButtonNavBar onPress={ navigation.state.params.handleThis} icon="mi-my-location" />,
```

Reply:
?

Reply:
yea, this isnt working. unsure how to fix.

Reply:
the original function looks correct. i get onPress is undefined error if i do what you suggested